File: d01/ex02/oddeven.js
var readline = require('readline')
var rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false
});

function odd_even (x) {
	if (x % 2 == 0)
		return 1;
	else
		return 0;
}
if (process.argv.length < 3)
	return ;

process.stdout.write("Enter a number: ")
rl.on('line', (line) => {
	var num = Number(line);
	if (isNaN(num) || line.length == 0)
		console.log(line + " is not a number")
	else if (odd_even(num) == 1)
		console.log("the number " + line + " is even");
	else
		console.log("the number " + line + " is odd");
	process.stdout.write("Enter a number: ")
});

File: d01/golang/ex02/oddeven.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	for {
		fmt.Printf("Enter a number: ")
		text, _ := reader.ReadString('\n')
		if text[0] == 27 {
			return  }
		i, err := strconv.ParseInt(strings.TrimSpace(text), 0, 64)
		if err != nil {
			fmt.Printf("%s is not a number", text)
		} else if i % 2 == 0 {
			fmt.Printf("The number %d is even\n", i)
		} else {
			fmt.Printf("The number %d is odd\n", i)}
	}
}
